cdap-unit-test/src/test/java/co/cask/cdap/security/AuthorizationTest.java

```diff
@@ -1466,4 +1440,20 @@ private void setUpPrivilegeAndRegisterForDeletion(Principal principal,
       cleanUpEntities.add(privilege.getKey());
     }
   }
+
+  private void waitForStoppedPrograms(final ProgramManager programManager) throws Exception {
+    Tasks.waitFor(true, new Callable<Boolean>() {
+      @Override
+      public Boolean call() throws Exception {
+        List<RunRecord> runs = programManager.getHistory();
+        for (RunRecord meta : runs) {
+          if (meta.getStatus() == ProgramRunStatus.STARTING ||
+            meta.getStatus() == ProgramRunStatus.RUNNING) {
```

Comment:
Technically, yes. None of the programs are ever suspended though in this test, so it wouldn't make a difference. This was also added just for this test because a separate auth test namespace is created that is not cleaned up automatically by `TestBase`.